plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -116,4 +130,208 @@ describe('KubernetesProxy', () => {
     expect(response.status).toEqual(299);
     expect(response.body).toStrictEqual(apiResponse);
   });
+
+  it('should add a serviceAccountToken as a `Authorization` request headers if one isnt provided by the users request', async () => {
+    worker.use(
+      rest.get('https://localhost:9999/api/v1/namespaces', (req, res, ctx) => {
+        if (!req.headers.get('Authorization')) {
+          return res(ctx.status(401));
+        }
+
+        if (req.headers.get('Authorization') !== 'Bearer randomToken') {
+          return res(ctx.status(403));
+        }
+
+        return res(
+          ctx.status(200),
+          ctx.json({
+            kind: 'NamespaceList',
+            apiVersion: 'v1',
+            items: [],
+          }),
+        );
+      }),
+    );
+
+    clusterSupplier.getClusters.mockResolvedValue([
+      {
+        name: 'cluster1',
+        url: 'https://localhost:9999',
+        serviceAccountToken: 'randomToken',
+        authProvider: 'serviceAccount',
+      },
+    ] as ClusterDetails[]);
+
+    authTranslator.decorateClusterDetailsWithAuth.mockResolvedValue({
+      name: 'cluster1',
+      url: 'https://localhost:9999',
+      serviceAccountToken: 'randomToken',
+      authProvider: 'serviceAccount',
+    } as ClusterDetails);
+
+    const app = express().use('/mountpath', proxy.createRequestHandler());
+    const requestPromise = request(app)
+      .get('/mountpath/api/v1/namespaces')
+      .set(HEADER_KUBERNETES_CLUSTER, 'cluster1');
+
+    worker.use(rest.all(requestPromise.url, req => req.passthrough()));
+
+    const response = await requestPromise;
+
+    expect(response.status).toEqual(200);
+  });
```

Comment:
I'm afraid the body of this test is identical to the one below, but with `randomToken` instead of `catToken`?
```suggestion
```

Reply:
Duplicated and redundant tests were removed